Fix RNG used by cosmetics editor to use same RNG method as rando

leggettc18 opened the pull request:
Apparently, `std::random_device` is not properly non-deterministic on certain situations where rng hardware is not accessible or something like that. I recently bumped into this with a Linux distro on my gaming desktop. So I essentially copied the randomizer rng to ShipUtils so I could use it for the Cosmetics Editor (with `std::random_device` it was always generating every color to the same dark blue color). It is a copy of the same functionality, it does not use the same seeds/state variables so randomizing cosmetics won't effect randomizer seed generation and vice/versa.

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/5071134220.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/5071134830.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/5071264531.zip)
<!--- section:artifacts:end -->

leggettc18 committed:
Fix RNG used by cosmetics editor to use same RNG method as rando

leggettc18 committed:
clang-format

serprex commented on soh/soh/ShipUtils.cpp:
```diff
@@ -96,3 +97,53 @@ extern "C" void* Ship_GetCharFontTexture(u8 character) {
+    state = seed;
+}
+
+uint32_t ShipUtils_next32() {
```
ideally random.cpp core functions would change to taking state as parameter, then these endpoints with globals could pass those into the core function

leggettc18 replied:
Yeah if we want to shift Rando over to these funcs as well that's what I would do, trying to decide if that's in-scope for right now or not. I guess if I'm doing this on develop instead of dev-copper I might as well go all the way.

serprex commented on soh/soh/ShipUtils.cpp:
```diff
@@ -96,3 +97,53 @@ extern "C" void* Ship_GetCharFontTexture(u8 character) {
+#if !defined(__SWITCH__) && !defined(__WIIU__)
+        uint64_t seed = static_cast<uint64_t>(std::random_device{}());
+#else
+        uint64_t seed = static_cast<uint64_t>(std::hash<std::string>{}(std::to_string(rand())));
```
```suggestion
        uint64_t seed = static_cast<uint64_t>(rand());
```
Beyond this PR's scope, but we should go ahead & rely on rand without the string-then-hash song & dance

leggettc18 replied:
If this was on dev-copper instead of dev I'd say it's probably out of scope but since its on dev I might as well go ahead and make the bigger changes, so I'm thinking this is maybe not quite out of scope after all.

leggettc18 committed:
Merge branch 'develop' into fix/rng/shiputils-rng

leggettc18 committed:
Refactor ShipUtils to optionally take a state pointer.

Also changed random.h/cpp to be a wrapper around ShipUtils RNG providing a pointer to the rando state variable.

leggettc18 commented:
@serprex I applied the refactor we discussed. ShipUtils RNG has a default rng state but can now take a pointer to a separate one, and the functions randomizer uses are now wrappers around ShipUtils RNG that just provide a pointer to a separate state variable for randomizer.

leggettc18 committed:
Use ShipUtils RNG for UIWidgets GetRandomValue

leggettc18 committed:
Fix AudioEditor to use ShipUtils RNG and its own state

It seems like shuffling Audio at the right time could have potentially messed with rando seed generation before this, but that bug, if it existed, should also be fixed with this.

serprex committed:
replace hash(str())
